packages/legend-query/src/components/QueryBuilderSetupPanel.tsx

```diff
@@ -79,6 +81,10 @@ const generateClassLabel = (
     val,
     queryBuilderState.graphManagerState.graph,
   );
+  let isDeprecatedClass = false;
+  if (val.stereotypes.find((st) => st.value.value === DEPRECATED_STEREOTYPE)) {
```

Comment:
Still think there is an issue here. 
You are only checking the stereotype value. 
Could be 
`meta::other::profile::OTHERPROFILE.deprecated`

Correct isEqual should be `graph.getProfile({PureProfile}).getStereotype(Deprecated) === st.value`.
one could argue we could have a `pureProfile` getter on PureModel for the first part